test/test_round.cpp

```diff
@@ -167,12 +179,26 @@ template <class T, class U>
 void check_modf_result(T a, T fract, U ipart)
 {
    BOOST_MATH_STD_USING
-   if (fract + ipart != a)
+
+   const T sum { fract + ipart };
+
+   #if defined(TEST_CPP_DOUBLE_FLOAT)
+   const T ratio { sum / a };
+
+   const T delta { fabs(1 - ratio) };
+   #endif
+
+   #if defined(TEST_CPP_DOUBLE_FLOAT)
```

Comment:
My only concern here, is we may have multiple TEST_* macros defined: this doesn't happen when running the Jamfile, but it is how I test locally ;)

Reply:
I am aware of this.

We could consider to query if `TEST_CPP_DOUBLE_FLOAT` is defined. And then combine this with the query if _NONE_ of the other ones are defined.

I was hoping you would ask.

Then we (or I) could make a macro that says, hey if it is `TEST_CPP_DOUBLE_FLOAT` only, then do this. Otherwise don't.

But the other day, when looking at some setups, I thought this was already being done. But maybe I got confuesd and it is only for specfun. Let me check.